DatabaseAccess.cs

```diff
@@ -0,0 +1,112 @@
+            {
+                using (var tableCmd = connection.CreateCommand())
+                {
+                    connection.Open();
+                    tableCmd.CommandText =
+                        @"CREATE TABLE IF NOT EXISTS codingTracker (
+                    Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                    StartTime TEXT,
+                    EndTime TEXT,
+                    Duration TEXT
+                    )";
+
+                    tableCmd.ExecuteNonQuery();
+                }
+            }
+        }
+
+        internal static void UpdateTable(int id, string startTime, string endTime, string duration)
+        {
+            using (var connection = new SQLiteConnection(dbConnectionString))
+            {
+                connection.Open();
+                using (var command = new SQLiteCommand(connection))
+                {
+                    command.CommandText = "UPDATE codingTracker SET StartTime = @startTime, EndTime = @endTime, Duration = @duration WHERE Id = @id";
+                    command.Parameters.AddWithValue("@startTime", startTime);
+                    command.Parameters.AddWithValue("@endTime", endTime);
+                    command.Parameters.AddWithValue("@duration",duration);
+                    command.Parameters.AddWithValue("@id", id);
+                    command.Prepare();
+
+                    command.ExecuteNonQuery();
+                }
+            }
+        }
+
+        internal static void InsertTable(string duration, string startTime, string endTime)
+        {
+            using (var connection = new SQLiteConnection(dbConnectionString))
+            {
+                connection.Open();
+                using (var command = new SQLiteCommand(connection))
+                {
+                    command.CommandText = "INSERT INTO codingTracker (StartTime, EndTime, Duration) VALUES(@startTime, @endTime, @duration)";
+                    command.Parameters.AddWithValue("@startTime", startTime);
+                    command.Parameters.AddWithValue("@endTime", endTime);
+                    command.Parameters.AddWithValue("@duration", duration);
+                    command.Prepare();
+
+                    command.ExecuteNonQuery();
+                }
+            }
+        }
+
+        internal static void DeleteTable(int id)
+        {
+            using (var connection = new SQLiteConnection(dbConnectionString))
+            {
+                connection.Open();
+                using (var command = new SQLiteCommand(connection))
+                {
+                    command.CommandText = "DELETE FROM codingTracker WHERE Id = @id";
+                    command.Parameters.AddWithValue("@id", id);
+                    command.Prepare();
+
+                    command.ExecuteNonQuery();
+                }
+            }
+        }
+
+        internal static void ViewTable()
+        {
+            Controller.CodingController.table.Clear();
```

Comment:
I think you can remove the "Controller" reference from here and add it as an using statement instead.